pymatgen/analysis/xas/spectrum.py

```diff
@@ -2,57 +2,195 @@
 # Copyright (c) Pymatgen Development Team.
 # Distributed under the terms of the MIT License.
 
-from pymatgen.core.spectrum import Spectrum
-import numpy as np
-
 """
-This module defines classes to represent all xas
+This module defines classes to represent all xas and stitching methods
 """
+import math
+from scipy.interpolate import interp1d
+import numpy as np
+from pymatgen.analysis.structure_matcher import StructureMatcher
+from pymatgen.core.spectrum import Spectrum
 
-__author__ = "Chen Zheng"
+__author__ = "Chen Zheng, Yiming Chen"
 __copyright__ = "Copyright 2012, The Materials Project"
-__version__ = "2.0"
-__maintainer__ = "Chen Zheng"
-__email__ = "chz022@ucsd.edu"
-__date__ = "Aug 9, 2017"
+__version__ = "3.0"
+__maintainer__ = "Yiming Chen"
+__email__ = "chz022@ucsd.edu, yic111@ucsd.edu"
+__date__ = "Jan 26, 2020"
 
 
-class XANES(Spectrum):
+class XAS(Spectrum):
     """
-    Basic XANES object.
+    Basic XAS object.
 
     Args:
         x: A sequence of x-ray energies in eV
         y: A sequence of mu(E)
-        structure (Structure): Structure associated with the XANES.
-        absorption_specie (Specie): Specie associated with the XANES.
-        edge (str):
+        structure (Structure): Structure associated with the spectrum
+        absorption_element (Element): Element associated with the spectrum
+        edge (str): Absorption edge associated with the spectrum
+        spectrum_type (str): 'XANES' or 'EXAFS'
+
 
     .. attribute: x
         The sequence of energies
 
     .. attribute: y
         The sequence of mu(E)
 
-    .. attribute: absorption_specie
-        The absorption_species of the spectrum
+    .. attribute: absorption_element
+        The absorption_element of the spectrum
 
     .. attribute: edge
-        The edge of XANES spectrum
+        The edge of the spectrum
+
+    .. attribute: spectrum_type
+        XANES or EXAFS spectrum
+
     """
     XLABEL = 'Energy'
     YLABEL = 'Intensity'
 
-    def __init__(self, x, y, structure, absorption_specie, edge):
-        super().__init__(x, y, structure, absorption_specie, edge)
+    def __init__(self, x, y, structure, absorption_element, edge="K",
+                 spectrum_type="XANES"):
+        """
+        Initializes a spectrum object.
+        """
+
+        super(XAS, self).__init__(x, y, structure, absorption_element,
+                                  edge)
         self.structure = structure
-        self.absorption_specie = absorption_specie
+        self.absorption_element = absorption_element
```

Comment:
Can we change this to `absorbing_element`?

Reply:
I already changed absorption_specie to absorbing_element because usually oxidation states are not known so absorbing_element is a more accurate way to describe the x-ray absorption process.

Reply:
You have absor**ption**_element right now, and I'd prefer it if it was absor**bing**_element: switching from "ption" to "bing" which is just a bit clearer. It is a bit pedantic, but might as well take care of it now.

Reply:
Thanks for suggestion. Just renamed it.